src/towncrier/_settings/load.py

```diff
@@ -107,52 +100,70 @@ def load_config_from_file(directory: str, config_file: str) -> Config:
 
 
 def parse_toml(base_path: str, config: Mapping[str, Any]) -> Config:
-    if "tool" not in config:
+    if "towncrier" not in (config.get("tool") or {}):
         raise ConfigError("No [tool.towncrier] section.", failing_option="all")
 
     config = config["tool"]["towncrier"]
+    parsed_data = {}
+
+    # Check for misspelt options.
+    for typo, correct in [
+        ("singlefile", "single_file"),
+    ]:
+        if config.get(typo):
+            raise ConfigError(
+                f"`{typo}` is not a valid option. Did you mean `{correct}`?",
+                failing_option=typo,
+            )
 
-    sections = OrderedDict()
+    # Process options.
+    for field in dataclasses.fields(Config):
+        if field.name in ("sections", "types", "template"):
+            # Skip these options, they are processed later.
+            continue
+        if field.name in config:
+            #  Interestingly, the __future__ annotation turns the type into a string.
```

Comment:
ah yeah, attrs has https://www.attrs.org/en/latest/api.html#attrs.resolve_types for that. I thought DCs had something too…